### PR TITLE
Add base implementation СRUD operations delete, insert, replace

### PR DESCRIPTION
**Pull request by savolgin**
https://www.notion.so/CRUD-094381826d32481a8b604ea45ad5cfb5

### CODE REVIEW

File: src/main/java/io/tarantool/driver/api/tuple/TarantoolUpdateTuple.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        this.op = op;
+        this.fieldPosition = fieldPosition;
+        this.argument = argument;
+    }
+
+    public List<Object> toArray() {
+        MessagePackMapper mapper = DefaultMessagePackMapperFactory.getInstance().defaultComplexTypesMapper();
+        return Collections.singletonList(
+                Arrays.asList(
+                        new TarantoolFieldImpl<>(op, mapper),
```
Comment (akudiyar) [inline src/main/java/io/tarantool/driver/api/tuple/TarantoolUpdateTuple.java:31]: I think TarantoolFieldImpl is not necessary here, we can use fieldimpl internals and implement the `Packable` interface

**Comment by akudiyar** [comment]
I suggest separating update and upsert requests into a new PR, since it is a separate task

File: src/main/java/io/tarantool/driver/protocol/requests/TarantoolDeleteRequest.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+ * Delete request.
+ * See <a href="https://www.tarantool.io/en/doc/2.3/dev_guide/internals/box_protocol/#binary-protocol-requests">
+ *     https://www.tarantool.io/en/doc/2.3/dev_guide/internals/box_protocol/#binary-protocol-requests</a>
+ *
+ * @author Sergey Volgin
+ */
+public final class TarantoolDeleteRequest extends TarantoolRequest {
+
+    /**
+     * (non-Javadoc)
```
[the remark this answers is not in the document]
Comment (savolgin) [reply src/main/java/io/tarantool/driver/protocol/requests/TarantoolDeleteRequest.java]: done

File: src/main/java/io/tarantool/driver/protocol/requests/TarantoolDeleteRequest.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+     */
+    public static class Builder {
+
+        Map<Integer, Object> bodyMap;
+
+        public Builder() {
+            this.bodyMap = new HashMap<>(3, 1);
+        }
+
+        public Builder withSpaceId(int spaceId) {
```
[the remark this answers is not in the document]
Comment (savolgin) [reply src/main/java/io/tarantool/driver/protocol/requests/TarantoolDeleteRequest.java:45]: added

File: src/main/java/io/tarantool/driver/mappers/DefaultArrayValueToTarantoolTupleConverter.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+import io.tarantool.driver.api.tuple.TarantoolTuple;
+import io.tarantool.driver.api.tuple.TarantoolTupleImpl;
+import org.msgpack.value.ArrayValue;
+
+/**
+ * Default {@link ArrayValue} to {@link TarantoolTuple} converter
+ *
+ * @author Sergey Volgin
+ */
+public class DefaultArrayValueToTarantoolTupleConverter  implements ValueConverter<ArrayValue, TarantoolTuple> {
```
Comment (akudiyar) [inline src/main/java/io/tarantool/driver/mappers/DefaultArrayValueToTarantoolTupleConverter.java:12]: Rename to `DefaultTarantoolTupleValueConverter` (analogous to `DefaultListValueConverter`, which converts ArrayValue to List)

File: src/main/java/io/tarantool/driver/mappers/DefaultMessagePackMapper.java [hunk 5]
```diff
@@ -289,6 +290,15 @@ public Builder withDefaultListObjectConverter() {
             return this;
         }
 
+        /**
+         * Configure the mapper with default {@code MP_ARRAY} entity to {@link TarantoolTuple} converter
+         * @return builder
+         */
+        public Builder withDefaultArrayValueToTarantoolTupleConverter() {
```
Comment (akudiyar) [inline src/main/java/io/tarantool/driver/mappers/DefaultMessagePackMapper.java:297]: I think we shouldn't have such method here, let's add it to `TarantoolResultMapperFactory`. This mapper deals only with primitive types, and TarantoolTuple is only one variant of a complex type representation

File: src/test/java/io/tarantool/driver/integration/StandaloneTarantoolClientIT.java [hunk 6]
```diff
@@ -1,45 +1,268 @@
...
+                new ImmutableLongValueImpl(1968L)
+        );
+
+        TarantoolTuple tarantoolTuple = new TarantoolTupleImpl(arrayValue, mapperFactory.defaultComplexTypesMapper());
+        testSpace.insert(tarantoolTuple, v -> v);
+
+        TarantoolIndexQuery query = new TarantoolIndexQuery();
+        query.withKeyValues(Collections.singletonList(123));
+
+        TarantoolUpdateTuple updateTuple = new TarantoolUpdateTuple("+", 3, 100000);
```
[the remark this answers is not in the document]
Comment (akudiyar) [reply src/test/java/io/tarantool/driver/integration/StandaloneTarantoolClientIT.java]: Suggestions were moved to the new task description. Will be applied in a separate PR

File: src/main/java/io/tarantool/driver/mappers/TarantoolResultMapper.java [hunk 7]
```diff
@@ -21,11 +23,21 @@
...
+    @SuppressWarnings("unchecked")
     public TarantoolResultMapper(MessagePackValueMapper valueMapper,
                                  ValueConverter<ArrayValue, T> tupleConverter) {
         this.valueMapper = valueMapper;
-        valueMapper.registerValueConverter(ArrayValue.class, TarantoolResultImpl.class,
+        valueMapper.registerValueConverter(ArrayValue.class, TarantoolResult.class,
                 v -> new TarantoolResultImpl<>(v, tupleConverter));
+
+        try {
+            Class<T> type = MapperReflectionUtils.getConverterTargetType(tupleConverter);
```
Comment (akudiyar) [inline src/main/java/io/tarantool/driver/mappers/TarantoolResultMapper.java:34]: I didn't suppose it to work like that. I think that TarantoolTuple converter should be added in factory

File: src/main/java/io/tarantool/driver/mappers/DefaultMessagePackMapper.java [hunk 8]
```diff
@@ -1,5 +1,6 @@
 package io.tarantool.driver.mappers;
 
+import io.tarantool.driver.api.tuple.TarantoolTuple;
```
Comment (akudiyar) [inline src/main/java/io/tarantool/driver/mappers/DefaultMessagePackMapper.java:3]: extra import

File: src/main/java/io/tarantool/driver/mappers/TarantoolResultMapper.java (hunk 7 above)
Comment (akudiyar) [reply src/main/java/io/tarantool/driver/mappers/TarantoolResultMapper.java:34]: Adding TarantoolTuple mapper here seems to be not necessary

Comment (savolgin) [reply src/main/java/io/tarantool/driver/mappers/TarantoolResultMapper.java]: removed

File: src/main/java/io/tarantool/driver/mappers/DefaultMessagePackMapper.java (hunk 8 above)
Comment (savolgin) [reply src/main/java/io/tarantool/driver/mappers/DefaultMessagePackMapper.java]: removed

File: src/main/java/io/tarantool/driver/TarantoolClientConfig.java [hunk 9]
```diff
@@ -95,35 +92,19 @@ public void setReadTimeout(int readTimeout) {
...
-    public void setObjectMapper(MessagePackObjectMapper objectMapper) {
-        this.objectMapper = objectMapper;
+    public MessagePackMapper getMessagePackMapper() {
+        return messagePackMapper;
     }
 
     /**
-     * Get MessagePack entity-to-object mapper
-     * @return a {@link MessagePackObjectMapper} instance
+     * Get mapper between Java objects and MessagePack entities
```
Comment (akudiyar) [inline src/main/java/io/tarantool/driver/TarantoolClientConfig.java:103]: Set*

Comment (akudiyar) [reply src/main/java/io/tarantool/driver/TarantoolClientConfig.java:103]:
```suggestion
     * Set mapper between Java objects and MessagePack entities
```

**Pull request merged**